### DISCUSSION TITLE
Loss on torch_geometric.nn.dense_diff_pool

### DISCUSSION THREAD

**Discussion by Hu-Chuxuan** [pyg-team/pytorch_geometric#4832] [Q&A]
Hello~ I am quite puzzled about the loss generated from this matrix, specifically, I believe the loss should not have divided by adj.numel() here.
<img width="262" alt="1655760563(1)" src="https://user-images.githubusercontent.com/81068196/174680019-bac480f9-23ad-40c0-8403-538c8d039eea.png">
From the original paper, this term is simply the Frobenius norm of A and S(l)S(l)
<img width="288" alt="1655760706(1)" src="https://user-images.githubusercontent.com/81068196/174680250-2e3aa8cb-f03c-40c3-8967-3530f0883cf5.png">
If I used the link loss generated from torch_geometric.nn.dense_diff_pool, it will be very small and the eventual coarsened outcome will not be similar to original graph(i.e. most nodes will be likely to gather in one cluster). After I multiply the adj.numel() back, it became normal.
I am not sure about why you divide that adj.numel(), is it due to other considerations?

**Comment by rusty1s** [comment]
This is directly taken from here: https://github.com/RexYing/diffpool/blob/8dfb97cf60c2376ac804761837b9966f1d302acb/encoders.py#L397. It is used to normalize the loss in order for it to be agnostic to graph sizes.
